### PR TITLE
please pull

### PR DESCRIPTION
**Pull request by jonassmedegaard**

### CODE REVIEW

**Comment by ironcamel** [comment]
Thanks for the patch. Please make one change. I would prefer an entry to be a subresource of a feed. For example:

```
get '/feeds/:feed_title/entries/:entry_id' => sub { ... }
```

or

```
get '/feeds/:feed_title/:entry_id' => sub { ... }
```

I think the first way is more flexible in case we wanted wanted to extend it by adding more subresources, e.g., '/feeds/:feed_title/subscribers'.

**Comment by jonassmedegaard** [comment]
I want the route stems to be configurable - also supporting joining them like this.  But sure, I don't mind having them joined by default.

**Comment by jonassmedegaard** [comment]
Whoops!

Please don't pull yet - I discovered flaws in it.

Am writing regression tests now!

**Comment by jonassmedegaard** [comment]
Ok - good to go now, I believe!

File: lib/AtomBus.pm [hunk 1]
```diff
@@ -112,12 +139,18 @@ post '/feeds/:feed_title' => sub {
...
+    _add_etag($entry->id);
+    header Location => $entry->link->href;
+    content_type 'application/atom+xml;type=entry';
+    status 'created';
+    return $entry->as_xml;
 };
 
 sub _gen_id { 'urn:uuid:' . create_UUID_as_string() }
 
+sub _id_nss { $_ = shift; s/^urn:uuid://; return $_ }
```
Comment (ironcamel) [inline lib/AtomBus.pm:152]: What does nss stand for?

Comment (jonassmedegaard) [inline lib/AtomBus.pm:152]: On 11-06-15 at 05:55pm, ironcamel wrote:

> >  };
> > 
> >  sub _gen_id { 'urn:uuid:' . create_UUID_as_string() }
> > 
> > +sub _id_nss { $_ = shift; s/^urn:uuid://; return $_ }
> 
> What does nss stand for?

Namespace-Specific String.

Documented at urn: section of manpage for Perl module URI.
- Jonas

## 
- Jonas Smedegaard - idealist & Internet-arkitekt
- Tlf.: +45 40843136  Website: http://dr.jones.dk/
  
  [x] quote me freely  [ ] ask before reusing  [ ] keep private

Comment (jonassmedegaard) [inline lib/AtomBus.pm:152]: On a related note, I would prefer to store only bare nss in database, and have thought about ways to support both for some transition period. I find it suboptimal to hardcode scheme + Namespace Identifier as it is today.

For now I've done the less controversial of not touching existing use of UUID and only made elegant the newly introduced http URLs for entries.

Comment (ironcamel) [inline lib/AtomBus.pm:152]: I would be open to just storing the nss in the db. In a separate pull request please.

Comment (jonassmedegaard) [inline lib/AtomBus.pm:152]: On 11-06-15 at 07:00pm, ironcamel wrote:

> >  };
> > 
> >  sub _gen_id { 'urn:uuid:' . create_UUID_as_string() }
> > 
> > +sub _id_nss { $_ = shift; s/^urn:uuid://; return $_ }
> 
> I would be open to just storing the nss in the db. In a separate pull 
> request please.

Certainly :-)

## 
- Jonas Smedegaard - idealist & Internet-arkitekt
- Tlf.: +45 40843136  Website: http://dr.jones.dk/
  
  [x] quote me freely  [ ] ask before reusing  [ ] keep private

**Pull request merged**